### PR TITLE
Fix `Currency` backward compability broken bug 

### LINKED ISSUE #3268: `Currency`'s backward-compatibility was broken
Since [2.0.0](https://github.com/planetarium/libplanet/releases/tag/2.0.0), the `Currency` serialization's backward compatibility was broken. So `TxExecution`s created before 2.0.0, cannot be deserialized because their own Currencies cannot be deserialized.

image

### PR DESCRIPTION
**Pull request by riemannulus**
closed #3268

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -95,6 +96,7 @@ be compatible with this version.
 [#3274]: https://github.com/planetarium/libplanet/pull/3274
 [#3282]: https://github.com/planetarium/libplanet/pull/3282
 [#3283]: https://github.com/planetarium/libplanet/pull/3283
+[#3284]: https://github.com/planetarium/libplanet/pull/3284
```
Comment (moreal) [inline CHANGES.md:99]: Could you reference https://github.com/planetarium/libplanet/issues/3280 too?

File: CHANGES.md [hunk 2]
```diff
@@ -83,6 +83,7 @@ be compatible with this version.
     when messages are received.  [[#3273]]
 
 ### Bug fixes
+ -  Fixed a broken backwards compatibility of Currency  [[#3284]]
```
Comment (moreal) [inline CHANGES.md:86]: Could you reference https://github.com/planetarium/libplanet/issues/3268 too?

Comment (moreal) [inline CHANGES.md:86]:
```suggestion
 -  Fixed a broken backwards compatibility of Currency  [[#3268], [#3284]]
```

I couldn't suggest on not edited lines. You should manually add the below line.

```
[#3268]: https://github.com/planetarium/libplanet/issues/3268
```

image

**Comment by greymistcube** [comment]
Eh, seems like `TxSuccess` was the culprit, and we might have `IStore` already corrupted with two different serialized `Currency` formats. 😕

**Comment by riemannulus** [comment]
> Eh, seems like TxSuccess was the culprit, and we might have IStore already corrupted with two different serialized Currency formats. 😕

Yes, so I try to adapt both cases. (https://github.com/planetarium/libplanet/pull/3284/files#diff-4ba5d5e78e585f64e217441ed95e634f2d1b5c5df16ac3d419834d040882d7c4R115-R126)

**Review by moreal — APPROVED** [review]

**Review by moreal — APPROVED** [review]

**Pull request merged**